tools/azure-sdk-tools/packaging_tools/generate_sdk.py

```diff
@@ -18,10 +18,11 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def generate(config_path, sdk_folder, project_pattern, readme, restapi_git_folder, autorest_bin=None, force_generation=False):
+def generate(config_path, sdk_folder, project_pattern, readme, restapi_git_folder, autorest_bin=None, force_generation=False, config=None):
```

Comment:
Why do we need pass `config` as a parameter? It seems `config = read_config(sdk_folder, config_path)` is the same with what you pass in.

Reply:
Can we just return `config` instead of passing `config` as a parameter to avoid side effect?